community/kernel/src/main/java/org/neo4j/kernel/impl/api/schema/NodeSchemaMatcher.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.api.schema;
+
+import java.util.Iterator;
+
+import org.neo4j.collection.primitive.Primitive;
+import org.neo4j.collection.primitive.PrimitiveIntSet;
+import org.neo4j.kernel.api.schema_new.LabelSchemaDescriptor;
+import org.neo4j.kernel.api.schema_new.SchemaDescriptor;
+import org.neo4j.kernel.impl.api.KernelStatement;
+import org.neo4j.kernel.impl.api.operations.EntityReadOperations;
+import org.neo4j.storageengine.api.NodeItem;
+
+/**
+ * This class holds functionality to match LabelSchemaDescriptors to nodes
+ * @param <SUPPLIER> the type to match. Must implement LabelSchemaDescriptor.Supplier
+ */
+public class NodeSchemaMatcher<SUPPLIER extends LabelSchemaDescriptor.Supplier>
+{
+    private final EntityReadOperations readOps;
+
+    public NodeSchemaMatcher( EntityReadOperations readOps )
+    {
+        this.readOps = readOps;
```

Comment:
I didn't think much about that, as far as I can tell this follows the same path as before this PR. If that is broken we should look at it more and fix in separate PR.